datafusion/physical-plan/src/lib.rs

```diff
@@ -428,6 +428,18 @@ pub trait ExecutionPlan: Debug + DisplayAs + Send + Sync {
     fn statistics(&self) -> Result<Statistics> {
         Ok(Statistics::new_unknown(&self.schema()))
     }
+
+    /// Returns `true` if a limit can be safely pushed down through this
+    /// `ExecutionPlan` node.
+    fn supports_limit_pushdown(&self) -> bool {
```

Comment:
I found this name somewhat confusing as it implied to me it was reporting if `with_fetch` was implemented, where now I see it refers to if it is ok to push a limit through the node

Perhaps we could rename it to something different. Perhaps  `can_push_limit` or `preserves_limit` ? 

I don't feel strongly about this

Reply:
I am also not sure of the best name. If we find something that's clearly better, maybe we can change it in a follow-on before a public release.